lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -210,30 +238,88 @@ def run(self, **kwargs):
             'node_info_index',
             f'server_node>{self.machine_id}>type'
         )
+        if setup_type == None:
+            self.logger.error("Setup type is not set, please set the"
+            " setup type and try again, run following command to set"
+            " the setup type: 'cortx_config server config type <VM|HW>'"
+            )
+            raise RuntimeError("Could not find the setup type in conf store")
+
 
         if setup_type == "VM":
-            if name or storage_type or controller_type:
+            if (
+                name
+                or storage_type
+                or controller_type
+                or cvg
+                or data_devices
+                or metadata_devices
+            ):
                 if not self.enclosure_id:
                     self.enclosure_id = "enc_" + self.machine_id
                     self.refresh_key_map()
                     self.store_in_file()
-                    self.update_pillar_and_conf('enclosure_id', self.enclosure_id)
+                    self.update_pillar_and_conf(
+                        'enclosure_id',
+                        self.enclosure_id
+                    )
 
                 if name:
                     self.update_pillar_and_conf('name', name)
 
-                if storage_type:
+                if storage_type != "virtual":
+                    self.logger.error(
+                        "Storage type needs to be 'virtual' for VM")
+                    raise ValueError("Incorrect argument value provided")
```

Comment:
Since `storage_type` is optional param, if it is not explicitly provided, we can fetch VM/HW from Confstore (`server config` API). This should not be mandatory.

Reply:
yes, incorporated this scenario.